test/user-router-test.js

```diff
@@ -0,0 +1,130 @@
+require('../src/init');
+require('./prepare');
+const chai = require('chai');
+const server = require('../src/app');
+const chaiHttp = require('chai-http');
+const _ = require('lodash');
+
+chai.use(chaiHttp);
+const chaiHttpAgent = chai.request(server).keepOpen();
+
+const userMock = {
+    "id": "1",
+    "name": "test",
+    "dob": "1990.01.01",
+    "address": "shanghai",
+    "description": "A nodejs developer",
+};
+
+const userRouterURI = '/user';
+
+describe("User Router Test", function () {
+    describe("#Create", function () {
+        it("Create User - Success", function (done) {
+            chaiHttpAgent.post(userRouterURI).send(userMock).end((err, res) => {
+                chai.expect(res).to.have.status(200);
+                chai.expect(res.body).to.have.property('docs').length.greaterThanOrEqual(1);
+                chai.expect(res.body.docs[0].id, userMock.id);
+                done();
+            })
+        });
+        it("Create User - Fail - Duplicate userId", function (done) {
+            chaiHttpAgent.post(userRouterURI).send(userMock).end((err, res) => {
+                chai.expect(res).to.have.status(500);
+                done();
+            })
+        });
+        it("Create User - Fail - Without userId", function (done) {
+            let mockData = _.cloneDeep(userMock);
+            delete mockData.id;
+            chaiHttpAgent.post(userRouterURI).send(mockData).end((err, res) => {
+                chai.expect(res).to.have.status(400);
+                done();
+            })
+        });
+    });
+    describe("#Get", function () {
+        it("Get User - Success - By ID", function (done) {
+            chaiHttpAgent.get(userRouterURI).query({id: userMock.id}).end((err, res) => {
```

Comment:
Does it mean the `Get User` test cases are relying on the previous `Create User` test?

Reply:
Yes. The test cases are in the order `create -> get -> update -> delete` to make sure the mock data is reusable and will be cleaned at last, even though I am using memory mongo for testing.